components/agenda/src/Agenda.svelte

```diff
@@ -161,26 +169,48 @@
       themeUrl = _this.theme;
     }
 
+    if (!isValidTimezone(_this.timezone)) {
+      console.warn(`Invalid IANA time zone: ${_this.timezone}`);
+      _this.timezone = undefined;
+    } else if (
+      _this.timezone === Intl.DateTimeFormat().resolvedOptions().timeZone
+    ) {
+      _this.timezone = undefined;
+    }
+
```

Comment:
this can be refactored as they both set the timezone to the same value. This should be an empty string rather than undefined because its type is string
```suggestion
    if (_this.timezone !== ""  && (!isValidTimezone(_this.timezone) || _this.timezone === Intl.DateTimeFormat().resolvedOptions().timeZone)) {
      _this.timezone = "";
    }

```